pkg/validator/csv.go

```diff
@@ -0,0 +1,32 @@
+package validator
+
+import (
+	"bytes"
+	"io"
+
+	"encoding/csv"
+)
+
+// CsvValidator is used to validate a byte slice that is intended to represent a
+// CSV file.
+type CsvValidator struct{}
+
+// Validate checks if the provided byte slice represents a valid .csv file.
+// https://pkg.go.dev/encoding/csv
+func (csvv CsvValidator) Validate(b []byte) (bool, error) {
+	csvReader := csv.NewReader(bytes.NewReader(b))
+	csvReader.TrimLeadingSpace = true
+
+	for {
+		_, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			return false, err
```

Comment:
Does the CSV output return the line/column number in the error output? If not, is there a way that can be added though a custom error?

Reply:
Let me check up on this. If so will add this support.

Reply:
@kehoecj It returns this error in the standard error being returned for an illegal csv file. Sample
```parse error on line 1, column 20: bare " in non-quoted-field```

Reply:
I do not think we need to parse the error again. It already contains the details.